### PR TITLE
[ragel] Update to 6.10 and unbreak build

### PR DESCRIPTION
**Pull request by ebraminio**
The third patch seems no longer needed as https://github.com/ebraminio/ragel/commit/bc183f5a282f9f88906d65a42901649bd7e83760

### CODE REVIEW

File: ports/ragel/portfile.cmake [hunk 1]
```diff
@@ -23,12 +23,10 @@ vcpkg_apply_patches(
     PATCHES
         "${CMAKE_CURRENT_LIST_DIR}/0001-Remove-unistd.h-include-1.patch"
         "${CMAKE_CURRENT_LIST_DIR}/0002-Remove-unistd.h-include-2.patch"
-        "${CMAKE_CURRENT_LIST_DIR}/0003-Fix-rsxgoto.cpp-build.patch"
 )
 
 vcpkg_configure_cmake(
     SOURCE_PATH ${SOURCE_PATH}
-    GENERATOR "Visual Studio 14 2015"
```
Comment (ebraminio) [inline ports/ragel/portfile.cmake]: Not sure why this was here but it blocks 2017 builds.

File: ports/ragel/CMakeLists.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 cmake_minimum_required(VERSION 3.0)
-project(ragel CXX)
```
Comment (ebraminio) [inline ports/ragel/CMakeLists.txt]: This seems trouble making, not sure if it adds any value.

File: ports/ragel/portfile.cmake [hunk 3]
```diff
@@ -7,11 +7,11 @@
...
 include(vcpkg_common_functions)
-set(SOURCE_PATH ${CURRENT_BUILDTREES_DIR}/src/ragel-6.9)
+set(SOURCE_PATH ${CURRENT_BUILDTREES_DIR}/src/ragel-6.10)
 vcpkg_download_distfile(ARCHIVE
-    URLS "http://www.colm.net/files/ragel/ragel-6.9.tar.gz"
-    FILENAME "ragel-6.9.tar.gz"
-    SHA512 46886a37fa0b785574c03ba6581d99bbeaa11ca65cf4fdc37ceef42f4869bd695694cd69b4b974a25cf539f004cb106e3debda17fc26e1a9a6a4295992733dbd
+    URLS "http://www.colm.net/files/ragel/ragel-6.10.tar.gz"
+    FILENAME "ragel-6.10.tar.gz"
+    SHA512 6c1fe4f6fa8546ae28b92ccfbae94355ff0d3cea346b9ae8ce4cf6c2bdbeb823e0ccd355332643ea72d3befd533a8b3030ddbf82be7ffa811c2c58cbb01aaa38
```
Comment (KindDragon) [inline ports/ragel/portfile.cmake]: You can use new `vcpkg_from_github` instead https://github.com/Microsoft/vcpkg/blob/add92555384f59ab10b6f551db0b685f8e9e5323/ports/cpprestsdk/portfile.cmake#L3-L9

File: ports/ragel/portfile.cmake [hunk 4]
```diff
@@ -1,17 +1,9 @@
...
-#   TARGET_TRIPLET is the current triplet (x86-windows, etc)
-#   PORT is the current port name (zlib, etc)
-#   CURRENT_BUILDTREES_DIR = ${VCPKG_ROOT_DIR}\buildtrees\${PORT}
-#   CURRENT_PACKAGES_DIR  = ${VCPKG_ROOT_DIR}\packages\${PORT}_${TARGET_TRIPLET}
-#
-
 include(vcpkg_common_functions)
-set(SOURCE_PATH ${CURRENT_BUILDTREES_DIR}/src/ragel-6.9)
+set(SOURCE_PATH ${CURRENT_BUILDTREES_DIR}/src/ragel-6.10)
 vcpkg_download_distfile(ARCHIVE
```
Comment (ebraminio) [inline ports/ragel/portfile.cmake]: "vcpkg_from_github" sounds very cool but building this project from source and not dist tarball adds even more problems as ragel's source compile dependency to another similar project, colm, AFAIK.

Comment (KindDragon) [inline ports/ragel/portfile.cmake]: Yeah, sorry. I removed my comment after I realized it :)

Comment (ebraminio) [inline ports/ragel/portfile.cmake]: No problem :) I just wanted to document that anyway.

**Comment by alexkaratarakis** [comment]
Looks solid, thanks for the PR!

**Pull request merged**